components/director/internal/formationmapping/auth_middleware.go

```diff
@@ -279,7 +279,7 @@ func (a *Authenticator) isFormationAuthorized(ctx context.Context, formationID s
 	f, err := a.formationRepo.GetGlobalByID(ctx, formationID)
 	if err != nil {
 		if apperrors.IsNotFoundError(err) {
-			return false, http.StatusBadRequest, errors.Wrapf(err, "while getting formation with ID: %q globally", formationID)
+			return false, http.StatusNotFound, errors.Wrapf(err, "while getting formation with ID: %q globally", formationID)
```

Comment:
we'll return 404 status code and error but in the handler this error is only logged for traceability and another "generic" error is returned - `errors.Errorf("An unexpected error occurred while processing the request. X-Request-Id: %s", correlationID)` which seems a little strange
Same for the FA function/handler

Actually, I saw it's the same comment as the first one